src/main/java/dk/kb/labsapi/ImageExport.java

```diff
@@ -531,4 +533,226 @@ private double convertInch1200ToPixels(double inch1200Value){
         final int DPI = 300;
         return inch1200Value * DPI / 1200.0;
     }
+
+
+    /************************************* Streaming implementation **********************************************/
+
+    /**
+     * Get images of newspaper pages with given query present in text.
+     * @param query     to search for.
+     * @param startTime is the earliest boundary for the query.
+     * @param endTime   is the latest boundary for the query.
+     * @param max       number of documents to fetch.
+     * @param output    to write images to as one combined zip file.
+     */
+    public void streamFullpageFromQuery(String query, Integer startTime, Integer endTime, Integer max, OutputStream output, String exportFormat) throws IOException {
```

Comment:
_There's only two hard problems in computer science: Cache invalidation, naming things and one-off errors_.

The `stream`-part in `streamFullpageFromQuery` is very implementation specific, which can make sense in some cases but most of the time we are more interested in what happens at a higher level. The `FromQuery` part is a bit redundant as `query` is one of the parameters. So what is really the point of this method? Maybe `exportFullpage`?